skelebot/components/componentFactory.py

```diff
@@ -44,9 +49,20 @@ def __init__(self):
             sys.path.append(pluginsHome)
             for pluginName in os.listdir(pluginsHome):
                 if (pluginName[0] != "_"):
-                    module = importlib.import_module("{name}.{name}".format(name=pluginName))
-                    plugin = getattr(module, pluginName[0].upper() + pluginName[1:])
-                    self.COMPONENTS[pluginName.lower()] = plugin
+                    try:
+                        module = importlib.import_module("{name}.{name}".format(name=pluginName))
+                        plugin = getattr(module, pluginName[0].upper() + pluginName[1:])
+                        self.COMPONENTS[pluginName.lower()] = plugin
+                    except Exception:
+                        print(PLUGIN_WARNING.format(pluginName))
+
```

Comment:
More of a nice-to-have, but you could tell the user what the plugin exception was. Something like:
```
PLUGIN_WARNING = WARNING_HEADER + " | The {} plugin contains a {} ({}) - Adding plugin to Quarantine"
...
except Exception as e:
    print(PLUGIN_WARNING.format(pluginName, e.__class__.__name__, e)
```

Reply:
I went back-and-forth on whether or not to add the error message, but ultimately decided against it. My thinking is that If the person who uses the plugin is just an end-user of the plugin, not the developer, an error message like that would probably just look like noise to them. In the future it would be nice to have a way to verify a plugin with Skelebot though. Perhaps that's something we can add into the plugin component to add validation when installing the plugin.

Also, this change is more of a first-step and mainly aimed at preventing bad plugins from completely crashing Skelebot. We can iterate further on this later. There is a lot more we can do with plugins, validation, and error-handling.